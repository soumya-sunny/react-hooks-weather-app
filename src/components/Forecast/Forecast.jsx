import React, { useEffect } from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Day from '../Day/Day'
import { useSelector } from 'react-redux';
// import classes from '*.module.css';

// import { Container } from './styles';
const useStyles = makeStyles((theme) => ({
    root: {
        position: "relative",
        marginTop: '65px'
    }
}));


function DailyForecast() {

    const forecasts = useSelector(state => state?.forecasts||[])

    const getFormattedDate = (i) => {
        const today = new Date();
        const nextday = new Date(today.getFullYear(), today.getMonth(), today.getDate() + i);
        const options = { day: 'numeric',weekday: 'short' };
        return nextday.toLocaleDateString("en-US", options)
    }

    const classes = useStyles();
    return (
        <Grid container={true} className={classes.root} justify="center" spacing={2}>
            <Grid container justify="center">
                <Typography sm={12} md={10} lg={10} variant="h6" noWrap>
                    5 day Forecast
          </Typography>
            </Grid>
            <Grid container sm={12} md={10} lg={10} justify="center" direction="row">
                {forecasts.map((forecast, i) => {
                    const formattedDate = getFormattedDate(i); return (<Grid item sm={2} md={2} lg={2} data-testid="forecast-container">
                        <Day data={forecast.main} date={formattedDate} />
                    </Grid>)
                })}

            </Grid>
        </Grid>
    );
}

export default DailyForecast;