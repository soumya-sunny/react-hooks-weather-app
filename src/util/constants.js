export const forecastURL=`http://api.openweathermap.org/data/2.5/forecast?appid=${process.env.REACT_APP_API_KEY}&cnt=5&`;

//test