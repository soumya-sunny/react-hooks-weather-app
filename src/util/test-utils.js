// test-utils.js
import React from 'react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { render as rtlRender } from '@testing-library/react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import rootReducer, { initialState as reducerInitialState } from '../redux/reducers'

function render(
  ui,
  {
    initialState = reducerInitialState,
    store = createStore(rootReducer, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}


function renderWithRouter(
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState = reducerInitialState,
    store = createStore(rootReducer, initialState),
  } = {}
) {
  const Wrapper = ({ children }) => (
    <Router history={history}> <Provider store={store}>{children}</Provider></Router>
  )
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  }
}
// re-export everything
export * from '@testing-library/react'

// override render method
export { render, renderWithRouter }