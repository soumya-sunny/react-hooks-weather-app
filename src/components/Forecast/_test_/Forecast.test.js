
import React from 'react'
import { render, fireEvent, screen } from '../../../util/test-utils'


import Forecast from '../Forecast'


describe('Forecast data', () => {
    let wrapper;
    beforeEach(()=>
   wrapper= render(<Forecast />, {
        initialState: {
            forecasts: [
             {   main:{ temp_min: 28, temp_max: 30 }},
             {   main:{ temp_min: 28, temp_max: 30 }},
             {   main:{ temp_min: 28, temp_max: 30 }},
             {   main:{ temp_min: 28, temp_max: 30 }},
             {   main:{ temp_min: 28, temp_max: 30 }},
                ]
        },
    })
    )
    test('render', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('It should display forecast for 5 days', () => {
        expect(screen.queryAllByText(/28/).length).toBe(5)
    })


})
