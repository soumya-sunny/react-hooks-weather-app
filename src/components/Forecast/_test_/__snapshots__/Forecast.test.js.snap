// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast data render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
            lg="10"
            md="10"
            sm="12"
          >
            5 day Forecast
          </h6>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-sm-12 MuiGrid-grid-md-10 MuiGrid-grid-lg-10"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
          >
            <div
              class="MuiPaper-root wrapper MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
              >
                13 Thu
              </h5>
              <h6
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
              >
                30
                °
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap"
              >
                28
                °
              </h6>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
          >
            <div
              class="MuiPaper-root wrapper MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
              >
                14 Fri
              </h5>
              <h6
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
              >
                30
                °
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap"
              >
                28
                °
              </h6>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
          >
            <div
              class="MuiPaper-root wrapper MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
              >
                15 Sat
              </h5>
              <h6
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
              >
                30
                °
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap"
              >
                28
                °
              </h6>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
          >
            <div
              class="MuiPaper-root wrapper MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
              >
                16 Sun
              </h5>
              <h6
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
              >
                30
                °
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap"
              >
                28
                °
              </h6>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
          >
            <div
              class="MuiPaper-root wrapper MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
              >
                17 Mon
              </h5>
              <h6
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
              >
                30
                °
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap"
              >
                28
                °
              </h6>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
          lg="10"
          md="10"
          sm="12"
        >
          5 day Forecast
        </h6>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-sm-12 MuiGrid-grid-md-10 MuiGrid-grid-lg-10"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
        >
          <div
            class="MuiPaper-root wrapper MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
            >
              13 Thu
            </h5>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
            >
              30
              °
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap"
            >
              28
              °
            </h6>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
        >
          <div
            class="MuiPaper-root wrapper MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
            >
              14 Fri
            </h5>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
            >
              30
              °
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap"
            >
              28
              °
            </h6>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
        >
          <div
            class="MuiPaper-root wrapper MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
            >
              15 Sat
            </h5>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
            >
              30
              °
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap"
            >
              28
              °
            </h6>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
        >
          <div
            class="MuiPaper-root wrapper MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
            >
              16 Sun
            </h5>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
            >
              30
              °
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap"
            >
              28
              °
            </h6>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
        >
          <div
            class="MuiPaper-root wrapper MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
            >
              17 Mon
            </h5>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
            >
              30
              °
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap"
            >
              28
              °
            </h6>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
